server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -156,14 +170,16 @@ Map<Collector, Map<String, Long>> buildSliceLevelBreakdown() {
             // max slice end time across all timing types
             long sliceMaxEndTime = Long.MIN_VALUE;
             long sliceMinStartTime = Long.MAX_VALUE;
-            for (QueryTimingType timingType : QueryTimingType.values()) {
-                if (timingType.equals(QueryTimingType.CREATE_WEIGHT)) {
+
+            for (String timingType : timingMetrics) {
+                if (timingType.equals(QueryTimingType.CREATE_WEIGHT.toString())) {
                     // do nothing for create weight as that is query level time and not slice level
                     continue;
                 }
 
                 // for each timing type compute maxSliceEndTime and minSliceStartTime. Also add the counts of timing type to
                 // compute total count at slice level
+                final String timingTypeTimeKey = timingType;
```

Comment:
Wondering why do need to introduce `timingTypeTimeKey`? Could we just use `timingType`?